LibDeflate.lua

```diff
@@ -1860,13 +1991,12 @@ local function Deflate(configs, WriteBits, WriteString, FlushWriter, str
 			for i = block_end-32767, block_end do
 				string_table[j] = string_table[i-offset]
 				j = j + 1
-			end
-
+            end
 			for k, t in pairs(hash_tables) do
 				local tSize = #t
 				if tSize > 0 and block_end+1 - t[1] > 32768 then
 					if tSize == 1 then
-						hash_tables[k] = nil
+						--hash_tables[k] = nil -- seems to be causing problems
 					else
```

Comment:
Don't think this causes the problem

Reply:
I was getting an error about an invalid key to next() because the key was being deleted, and then accessed in the for loop. I usually fix this problem by making a new table of keys to set to nil, which I then loop over after accessing the first table.
    
    local tab, keys_to_delete = {}
    for k,v in pairs(tab) do
        if condition then
            table.insert(keys_to_delete, k)
        end
    end
    for _,k in pairs(keys_to_delete) do tab[k] = nil end

Something like that.

Reply:
This does not reaccess the table element after setting it to nil. Can you reproduce it in a standalone Lua interpreter?  I believe ComputerCraft uses LiaJit 2.0.3.  Can you reproduce it in Lua 2.0.3 outside ComputerCraft environment?

Reply:
> -- Sample code to remove keys whose value is 0
for k, v in t do
    if v == 0 then
        k[v] = nil
    end
end

Reply:
The code above is the standard way to conditionally remove the element in the table.
I suppose this is a ComputerCraft specific error? 
If that is the case, I could use your workaround by "by making a new table of keys to set to nil, which I then loop over after accessing the first table", but for the performance issue, this is only activated when ComputerCraft is detected. The origin code should be used otherwise.